src/mesh/generic/Boundary.cpp

```diff
@@ -46,6 +49,19 @@ Boundary::Boundary(Json::Value const& times, Json::Value const& loads)
                 });
 }
 
+Boundary::Boundary(Json::Value const& boundary, std::string const& name)
+{
+    if (boundary["GenerateType"] == "Sinusoidal")
+    {
+        generate_sinusoidal(boundary, name);
+    }
+    else // if (boundary["GenerateType"] == "UserDefined")
+    {
```

Comment:
This can be filled out and throw an error if not the type isn't "Sinusoidal" or "UserDefined".